pkg/destroy/gcp/dns.go

```diff
@@ -26,18 +31,28 @@ func (o *ClusterUninstaller) listDNSZones(ctx context.Context) (private *dnsZone
 	if o.NetworkProjectID != "" {
 		projects = append(projects, o.NetworkProjectID)
 	}
+	if o.PrivateZoneProjectID != "" {
+		projects = append(projects, o.PrivateZoneProjectID)
+	}
 
 	for _, project := range projects {
-		req := o.dnsSvc.ManagedZones.List(project).Fields("managedZones(name,dnsName,visibility),nextPageToken")
+		req := o.dnsSvc.ManagedZones.List(project).Fields("managedZones(name,dnsName,visibility,labels),nextPageToken")
 		err = req.Pages(ctx, func(response *dns.ManagedZonesListResponse) error {
 			for _, zone := range response.ManagedZones {
 				switch zone.Visibility {
 				case "private":
-					if o.isClusterResource(zone.Name) || (o.PrivateZoneDomain != "" && o.PrivateZoneDomain == zone.DnsName) {
-						private = &dnsZone{name: zone.Name, domain: zone.DnsName, project: project}
+					if o.PrivateZoneProjectID == "" || project != o.PrivateZoneProjectID {
+						// The private zone should exist in the private zone project if one was provided.
+						continue
+					}
```

Comment:
@barbacbd This check is new. Would you mind explaining why we need this check :thinking:?

Reply:
Resolved in the latest changes :D